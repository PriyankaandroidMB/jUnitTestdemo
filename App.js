import React from 'react';
import {View, Text, TextInput} from 'react-native';
export default class App extends React.Component {
  constructor() {
    super();
    this.state = {
      data: 'test',
    };
  }
  render() {
    return (
      <View style={{flex: 1}}>
        <TextInput
          testID={'username'}
          onChangeText={text => this.change(text)}
          style={{backgroundColor: 'grey', marginBottom: 15}}
          placeholder="Enter Username"
        />
      </View>
    );
  }
}
