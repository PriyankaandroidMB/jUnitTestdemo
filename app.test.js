import React from 'react';
import {View, Text, TextInput} from 'react-native';
import App from './App';
import renderer from 'react-test-renderer';
it('Render Correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
