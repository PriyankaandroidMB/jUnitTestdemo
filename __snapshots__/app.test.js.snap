// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Username"
    style={
      {
        "backgroundColor": "grey",
        "marginBottom": 15,
      }
    }
    testID="username"
  />
</View>
`;
